### DISCUSSION TITLE
Ensuring one deployment at a time

### DISCUSSION THREAD

**Discussion by nicogl** [fluxcd/flux2#1058] [Q&A] [answered]
I set up an ImageUpdateAutomation resource that allows automatic deployment (release updates) based on new docker images being uploaded to the registry. Everything is working as expected and a new release rollout is triggered in the Kubernetes cluster, but I'm wondering what happens if, by chance, an image is uploaded twice in a short period of time (say 1 minute). My concern is what happens if the second deployment is triggered while the first one is still ongoing...

Is there any Flux-specific toolkit feature to control this? Sorry if this question was answered in another thread.
If the answer is no, and some generic k8s can be used instead, any hint would be appreciated. It's not necessary the whole explanation of how to implement it but at least a headline on what to use to get some guidance.

Thanks!

**Comment by stefanprodan** [comment, marked as answer]
If you configure [health checks](https://toolkit.fluxcd.io/components/kustomize/kustomization/#health-assessment) for the updated deployments, kustomize-controller will make sure that the 2nd update will be applied only after the 1st one has been rolled out.
